### PR TITLE
Fix i.MX6ULEVK boot

### PR DESCRIPTION
**Pull request by MrVan**
The three patches is to
- fix the uart address, which will cause panic.
- Handle gic/csu in optee
- Update readme, drop the hacked code in uboot.

### CODE REVIEW

File: core/arch/arm/plat-imx/platform_config.h [hunk 1]
```diff
@@ -101,20 +101,16 @@
...
-#define DEVICE0_TYPE		MEM_AREA_IO_NSEC
-
-/*
- * console uart virtual address
- * The physical address is 0x2020000, we mapped it to 0x4020000,
- * see DEVICE0_VA_BASE and DEVICE0_PA_BASE
- */
-#define CONSOLE_UART_BASE		(0x4020000)
+#define CONSOLE_UART_BASE		(UART0_BASE)
 #define CONSOLE_UART_PA_BASE		(UART0_BASE)
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-imx/platform_config.h]: Why do we have both `CONSOLE_UART_BASE` and   `CONSOLE_UART_PA_BASE`, I'd prefer to remove the latter.

**Comment by MrVan** [comment]
Tags applied. Added a new patch to drop CONFIG_UART_PA_BASE.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by MrVan** [comment]
Updated with tags applied

**Comment by jforissier** [comment]
Rebased and meged. Thanks!

**Pull request closed without merging**